File: BLIND-75/Array/best-time-to-buy-sell.ts
function maxProfit(prices: number[]): number {
  let minPrice = prices[0];
  let profit = 0;
  
  for(let price of prices) {
      if (price <= minPrice) {
          minPrice = price;
      } else {
          profit = Math.max(profit, price - minPrice);
      }
  }
  
  return profit;
};

/**
  go through prices maintainig the minPrice profit,
  at each point you can set a new minPrice if pthe current price is less than the previous price
  otherwise we can check if the current possible profit is better than the overall profit
**/

File: BestTimeToBuySell.java
public class BestTimeToBuySell {
  class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int minSoFar = prices[0];
        for (int i = 0; i < prices.length; i++) {
            minSoFar = Math.min(minSoFar, prices[i]);
            max = Math.max(max, prices[i] - minSoFar);
        }
        return max;
    }
}
}
